CMakeLists.txt

```diff
@@ -212,7 +212,7 @@ elseif (("${CMAKE_C_COMPILER_ID}" STREQUAL "GNU") OR ("${CMAKE_C_COMPILER_ID}" S
 	set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -O3")
 
 	# enable somewhat modern C++
-	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
+	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=gnu++11")
```

Comment:
> Why can't codemp be C++?

OK, yes, strictly speaking it could.

If there is consensus among the maintainers that you want all the code to be C++, you could rename all the `.c` in codemp to `.cpp`, fix any code that is valid C but not valid C++, and then `q_shared.h` could make use of C++ features... but that seems a much, much larger change than is needed to fix the build.